src/Helpers/LocalMachineHelper.php

```diff
@@ -45,9 +47,26 @@ public function __construct(InputInterface $input, OutputInterface $output) {
    * @return bool
    */
   public function commandExists($command): bool {
+    if (array_key_exists($command, $this->installedBinaries)) {
+      return (bool) $this->installedBinaries[$command];
+    }
     $os_command = OsInfo::isWindows() ? ['where', $command] : ['which', $command];
     // phpcs:ignore
-    return $this->execute($os_command, NULL, NULL, FALSE)->isSuccessful();
+    $exists = $this->execute($os_command, NULL, NULL, FALSE)->isSuccessful();
```

Comment:
@grasmash this only checks if the command exists in the PATH. It will fail in two scenarios, which are common enough that I think we need to account for:

1. A command that exists relative to the current working directory, i.e. `vendor/bin/foo`. This should be easy to check for, just see if the command exists relative to the cwd.
2. A command that's defined as an alias, i.e. `alias composer='composer2'`. Use `command -v` instead of `which` to solve this. [reference](https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script)

None of this accounts for Windows (I'm not enough of an expert to say), and there are also some edge cases it may miss (if a command exists but isn't executable). I'm fine with burying our head in the sand on those.

These are all problems we faced previously on ADS CLI and BLT, FWIW. Makes me think PHP needs a common library to deal with this kind of thing properly.

Reply:
> A command that exists relative to the current working directory, i.e. vendor/bin/foo. This should be easy to check for, just see if the command exists relative to the cwd.

Even if it does exist, we'd need to actually call it using the directory path. That would require a lot of changes to many commands in ACLI. Seems like a separate PR to me.